### DISCUSSION TITLE
SOLVED   [monorepo] I can't import my components in its own stories, after migrating our monorepo from yarn to pnpm

### DISCUSSION THREAD

**Discussion by nirbG** [storybookjs/storybook#20848] [Help]
Hey guys,
My teams and I moved from yarn to pnpm. but I am facing issues to import our components in our stories.

to sum up, we have decided to create a design system using Lit and storybook.
We have a dedicated npm package for storybook and a dedicated npm package for each components.

```
|___ MONOREPO
   |___ apps
   |  |__ [...]
   |
   |___ packages
     |___ micro-front-ends
     |   |__ [...]
     |
     |___ UI library
        |___ storybook
        |   |__.stoybook
        |   |___docs
        |   |___package.json
        |
        |___ components
           |___ badge
           |    |___ stories
           |    |  |__ badge.stories.ts
           |    |__ badge.ts
           |    |__ package.json
           |
           |___ button
           |    |___ stories
           |    |  |__ button.stories.ts
           |    |__ button.ts
           |    |__ package.json
           |
           |___ text-field
```

Here is a example of our stories. in this case is the `badge.stories.ts`

```
import {TemplateResult} from 'lit';
import {html} from 'lit/static-html.js';

import '@blabla/wc-storybook-utils';

import '@blabla/wc-svg-icon';
import {unsafe_fui_svg_icon} from '@blabla/wc-svg-icon';

import '@blabla/wc-badge';
import {unsafe_fui_badge} from '@blabla/wc-fui-badge';

export default {
  title: 'Components/Badges',
  argTypes: {..}
}

export const Sandbox = (controls): TemplateResult => {
  return html`
    <story-sandbox>
      <${unsafe_fui_badge}
        .color=${'primary'}
        .clickable=${true}>
        <${unsafe_svg_icon}slot="prefix" .size=${2}.svg=${'add'}></${unsafe_fui_svg_icon}>
          ${controls.label}
      </${unsafe_fui_badge}>
    </story-sandbox>
  `;
};
```

With yarn, I was able to import @blabla/wc-badge from its own npm package.
it worked because it was a phantom dependencies.
But since we moved to pnpm it doesn't works anymore. Because, if I don't specify this npm package in the package.json I won't be able to import it or it not even added to the node module at all.

For the moment I tried :
- To replace npm package name from the import by the name of the file but it doesn't works, 
   - import '../Badge';
- I tried to play with the .npmrc but I never found the right solution
- I tried to replace the package name by the path to the dist of the component. it works but I really don't like this solution
   - import '../dist';
- I tried to search how people use storybook and pnpm but I don't find a lot of example or docuementation

Has anyone run into the same problem ?
Or do you know how to configure pnpm and storybook for this kind of architecture ?

**Comment by airtonix** [comment]
Few things: 

- why are you importing the package and then importing it again declaring a named export? is this a `Lit` thing? seems prone to all kinds of mysterious errors if you've got "module level logic executing". I do everything possible to avoid architecting a monorepo that requires this pattern.
- For the `badge.stories.ts` you should only import it from it's local import path, never the public import path.


I actually think you're issues are around the things we can't see: 

- how is Lit effecting this?
- why do you have that wierd import duplication?
- how is your `tsconfig.json` configured? 

For lols:
- recreate your mono repo using nx.dev and just follow how it wants you to set things up. see if you experience the same problems.
- just import things with one import statement instead of two

**Reply by nirbG** [reply]
It was totally my bad.

I wasn't able to import my component with relative path, because I was facing error with Lit decorator. I changed my ts.config according to Lit doc. But  I didn't know that storybook use babel.
And according to their doc I had to add some lines in the config to handle Lit decorators.

Since in storybook v7 we have to handle the babel config by ourself i fixed my issue by adding some lines in the babel config.

I'm sorry that I wasted your time
